pkg/datarepair/queue/queue.go

```diff
@@ -0,0 +1,82 @@
+// Copyright (C) 2018 Storj Labs, Inc.
+// See LICENSE for copying information.
+
+package queue
+
+import (
+	"encoding/binary"
+	"sync"
+	"time"
+
+	"github.com/golang/protobuf/proto"
+	"github.com/zeebo/errs"
+
+	"storj.io/storj/pkg/pb"
+	"storj.io/storj/storage"
+	"storj.io/storj/storage/redis"
+)
+
+//RepairQueue is the interface for the data repair queue
+type RepairQueue interface {
+	Add(qi *pb.InjuredSegment) (storage.Key, error)
+	Remove(date storage.Key) error
+	GetNext() (storage.Key, pb.InjuredSegment, error)
+	GetSize() (int, error)
+}
+
+//Queue implements the RepairQueue interface
+type Queue struct {
+	DB    storage.KeyValueStore
+	mutex *sync.Mutex
+}
+
+var (
+	queueError = errs.Class("data repair queue error")
+)
+
+//NewQueue returns a pointer to a new Queue instance with an initialized connection to Redis
+func NewQueue(address, password string, db int) (*Queue, error) {
+	rc, err := redis.NewClient(address, password, db)
+	if err != nil {
+		return nil, err
+	}
+	return &Queue{
+		DB:    rc,
+		mutex: &sync.Mutex{},
+	}, nil
+}
+
+//Add adds a repair segment to the queue
+func (q *Queue) Add(qi *pb.InjuredSegment) (storage.Key, error) {
+	q.mutex.Lock()
+	defer q.mutex.Unlock()
+	dateTime := make([]byte, binary.MaxVarintLen64)
+	binary.PutVarint(dateTime, time.Now().UnixNano())
```

Comment:
Using varint time encoding seems weird here? It won't give you right order when listing items. https://play.golang.org/p/Hnd3xR1Uzf7.

Also don't ignore `n` argument that `PutVarint` returns.

Reply:
what do you suggest here instead?

Reply:
You can use big endian encoding (e.g. https://play.golang.org/p/CEspuJdbRPK). There are other options, but this probably makes most sense. However, keep in mind that `/` has special meaning in `storage.Key` so it might cause problems further down the line.